### DISCUSSION TITLE
Changing Combobox's SelectedItem from view model doesn't work

### DISCUSSION THREAD

**Discussion by jeff9571** [AvaloniaUI/Avalonia#14763] [Q&A]
![image](https://github.com/AvaloniaUI/Avalonia/assets/35749706/813a3a1c-b413-4dc2-9068-ea22f2de0183)
The objects in Channels have a property Device which is from RxList, I set one of the Channels' Device in view model, but the UI didn't update.

**Comment by timunie** [comment]
1. ItemsSoruce should be OneWay binding (default)
2. SelectedItem must send PropertyChanged event in order to work properly.

**Reply by jeff9571** [reply]
1.I set the ItemsSource binding mode two-way, it still not work.
2.The SelecteItem is like : 
    private DeviceModel? _device;

    [JsonIgnore]
    public DeviceModel? Device
    {
        get => _device;
        set => this.RaiseAndSetIfChanged(ref _device, value);
    }

**Reply by timunie** [reply]
Hmmm try to reproduce in a simple sample and upload it here, I need to investigate on my own.

**Reply by jeff9571** [reply]
[ComboboxTest.zip](https://github.com/AvaloniaUI/Avalonia/files/14445826/ComboboxTest.zip)
Please have a look

**Reply by timunie** [reply]
hmmm no clue rn.

**Reply by IanRawley** [reply]
There was a thing where the order of binding mattered.  If you bound the SelectedItem before binding the ItemSource the control would try to set the SelectedItem, find it wasn't in the ItemSource (which hadn't been bound yet) and then set it to null.  I think a fix was merged, but I don't know if it's made it to release yet?

Anyway, try binding the ItemSource before the SelectedItem and see if that fixes it.

**Reply by jeff9571** [reply]
I've try the way you said,but it didn't work.

**Reply by timunie** [reply]
Interestingly that is only an issue when inside ItemsControl.

**Reply by jeff9571** [reply]
How to solve it? Will this get fixed in next release?

**Reply by timunie** [reply]
> How to solve it?

If I knew, I'd have told you how to. You can test nightly again if it is fixed there.

**Reply by timunie** [reply]
probably a bug, see: https://github.com/AvaloniaUI/Avalonia/issues/14814
